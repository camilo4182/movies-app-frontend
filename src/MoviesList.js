import React, { Component } from 'react';
import { Button, ButtonGroup, Container, Table } from 'reactstrap';
import AppNavbar from './AppNavBar';
import { Link } from 'react-router-dom';

class MoviesList extends Component {

    constructor(props) {
        super(props);
        this.state = {movies: []};
        this.remove = this.remove.bind(this);
    }

    componentDidMount() {
        fetch('/movies')
            .then(response => response.json())
            .then(data => this.setState({movies: data}));
    }

    async remove(id) {
        await fetch(`/movies/${id}`, {
            method: 'DELETE',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            }
        }).then(() => {
            let updatedMovies = [...this.state.movies].filter(i => i.id !== id);
            this.setState({movies: updatedMovies});
        });
    }

    render() {
        const {movies} = this.state;

        const moviesList = movies.map(movie => {
            return <tr key={movie.id}>
                <td fortesting='titel-movie' style={{whiteSpace: 'nowrap'}}>{movie.title}</td>
                <td>{movie.description}</td>
                <td>{movie.director}</td>
                <td>
                    <ButtonGroup>
                        <Button size="sm" color="primary" tag={Link} to={"/movies/" + movie.id}>Edit</Button>
                        <Button size="sm" color="danger" onClick={() => this.remove(movie.id)}>Delete</Button>
                    </ButtonGroup>
                </td>
            </tr>
        });

        return (
            <div>
                <AppNavbar/>
                <Container fluid>
                    <div className="float-right">
                        <Button color="success" tag={Link} to="/movies/new">Add Movie</Button>
                    </div>
                    <h3>Movies</h3>
                    <Table className="mt-4">
                        <thead>
                        <tr>
                            <th width="30%">Title</th>
                            <th width="30%">Description</th>
                            <th width="30%">Director</th>
                            <th width="40%">Actions</th>
                        </tr>
                        </thead>
                        <tbody>
                        {moviesList}
                        </tbody>
                    </Table>
                </Container>
            </div>
        );
    }
}
export default MoviesList;