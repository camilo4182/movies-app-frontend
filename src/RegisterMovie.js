import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { Button, Container, Form, FormGroup, Input, Label } from 'reactstrap';
import AppNavbar from './AppNavBar';

class RegisterMovie extends Component {

    emptyItem = {
        title: '',
        description: '',
        director: ''
    };

    constructor(props) {
        super(props);
        this.state = {
            item: this.emptyItem
        };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    };

    handleChange(event) {
        const target = event.target;
        const value = target.value;
        const name = target.name;
        let item = {...this.state.item};
        item[name] = value;
        this.setState({item});
    };

    async handleSubmit(event) {
        event.preventDefault();
        const {item} = this.state;
    
        await fetch('/movies', {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(item),
        });
        this.props.history.push('/movies');
    };

    render() {
        const {item} = this.state;
        const title = <h2>Add Client</h2>;
    
        return <div>
            <AppNavbar/>
            <Container>
                {title}
                <Form onSubmit={this.handleSubmit}>
                    <FormGroup>
                        <Label for="title">Title</Label>
                        <Input type="text" name="title" id="title"
                               onChange={this.handleChange} autoComplete="title"/>
                    </FormGroup>
                    <FormGroup>
                        <Label for="description">Description</Label>
                        <Input type="text" name="description" id="description"
                               onChange={this.handleChange} autoComplete="description"/>
                    </FormGroup>
                    <FormGroup>
                        <Label for="director">Director</Label>
                        <Input type="text" name="director" id="director"
                               onChange={this.handleChange} autoComplete="director"/>
                    </FormGroup>
                    <FormGroup>
                        <Button color="primary" type="submit">Register</Button>{' '}
                        <Button color="secondary" tag={Link} to="/movies">Cancel</Button>
                    </FormGroup>
                </Form>
            </Container>
        </div>
    }
}
export default withRouter(RegisterMovie);