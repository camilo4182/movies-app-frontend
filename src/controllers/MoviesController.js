import axios from 'axios';

export const MoviesController = {
    register(movie) {
        return axios({
            method: 'POST',
            baseURL: process.env.API,
            url: 'movies',
            data: movie
        });
    },
    update(movie_title){
        return axios({
            method: 'PUT',
            baseURL: process.env.API,
            url: `movies/${movie_title}`
        });
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'movies'
        });
    },
    delete(movie_title) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `movies/${movie_title}`
        });
    }
}

