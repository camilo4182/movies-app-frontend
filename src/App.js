import React, { Component } from 'react';
import './App.css';
import Home from './Home';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import MoviesList from './MoviesList';
import MovieEdit from "./MovieEdit";

class App extends Component {
  render() {
    return (
        <Router>
          <Switch>
            <Route path='/' exact={true} component={Home}/>
            <Route path='/movies' exact={true} component={MoviesList}/>
            <Route path='/movies/:id' component={MovieEdit}/>
          </Switch>
        </Router>
    )
  }
}

export default App;