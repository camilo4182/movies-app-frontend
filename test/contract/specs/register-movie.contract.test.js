import {provider} from './init-pact';
import {MoviesController} from '../../../src/controllers';
import {Matchers} from '@pact-foundation/pact';

describe('Movie service', () => {
    describe('When a request to register a movie is made', () => {
        const movie = {
            title: 'Matrix',
            description: "Desc 1",
            director: "Dir 1"
        }
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'register a movie',
                uponReceiving: 'a request to register a movie',
                withRequest: {
                    method: 'POST',
                    path: '/movies',
                    body: movie
                },
                willRespondWith: {
                    status: 201,
                    headers:{
                        'Content-Type': 'application/json'
                    },
                    body: Matchers.somethingLike(
                        {
                            title: Matchers.like('Matrix'),
                            description: Matchers.like("Desc 1"),
                            director: Matchers.like("Dir 1")
                        }
                    )
                }
            });
        });

        it('Then it should return the right data', async () => {
            const response = await MoviesController.register(movie);
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

        afterAll(async () => {
            await provider.finalize();
        })
    });
});