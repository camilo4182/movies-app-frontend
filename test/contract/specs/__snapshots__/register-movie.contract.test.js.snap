// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie service When a request to register a movie is made Then it should return the right data 1`] = `
Object {
  "description": "Desc 1",
  "director": "Dir 1",
  "title": "Matrix",
}
`;
